pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/WorkerQueryExecutor.java

```diff
@@ -100,38 +100,39 @@ public void runJob() {
   }
 
   // TODO: split this PhysicalPlanner into a separate module
+  // TODO: optimize this into a framework. (physical planner)
   private BaseOperator<TransferableBlock> getOperator(long requestId, StageNode stageNode,
       Map<Integer, StageMetadata> metadataMap) {
-    // TODO: optimize this into a framework. (physical planner)
     if (stageNode instanceof MailboxReceiveNode) {
       MailboxReceiveNode receiveNode = (MailboxReceiveNode) stageNode;
       List<ServerInstance> sendingInstances = metadataMap.get(receiveNode.getSenderStageId()).getServerInstances();
-      return new MailboxReceiveOperator(_mailboxService, RelDistribution.Type.ANY, sendingInstances, _hostName, _port,
-          requestId, receiveNode.getSenderStageId());
+      return new MailboxReceiveOperator(_mailboxService, receiveNode.getDataSchema(), RelDistribution.Type.ANY,
```

Comment:
Where do we do schema validation ? As in, if the receiver receives the DataBlock but the schema sent as part of it by the sender stage is not same as what receiver was expecting when stages were setup, then it should error out ?

Reply:
yeah. exactly